### PR TITLE
Improvements to decentralized coordination

### PR DESCRIPTION
**Pull request by edwardalee**
This PR fixes a number of issues with STA and STAA in decentralized coordination. There is a [companion PR in lingua-franca](https://github.com/lf-lang/lingua-franca/pull/2394). In particular:
* It enables using large values when the data flow is predictable. 
* It changes terminology to use STA and STAA.
* It guards against overflow of time values, allowing large STA and STAA.
* It ensures that STA is always added to STAA.
* It moves the logic of whether to add the STA out of wait_until and into the federated code.
* It makes sure the stop tag is completed before exiting.

### CODE REVIEW

**Review by Depetrol — APPROVED** [review]
I think this is a great improvement for decentralized coordination, using `STA` & `STAA` could really reduce a lot of the confusion about `STP`. I can confirm that this solves https://github.com/lf-lang/lingua-franca/issues/2373.

**Comment by lhstrh** [comment]
There are non-deterministic failures lurking somewhere. The CI run of https://github.com/lf-lang/reactor-c/pull/476/commits/21dee16522bcc9843dd0d18e9bca71bed3081ed4 failed, but the subsequent run on https://github.com/lf-lang/reactor-c/pull/476/commits/89b0e3617db1d2730552a418fd35f1395afe114a passed, even though the commit only includes comments...

**Comment by edwardalee** [comment]
> There are non-deterministic failures lurking somewhere. The [CI run](https://github.com/lf-lang/reactor-c/actions/runs/10254409515/job/28369131225) based [21dee16](https://github.com/lf-lang/reactor-c/commit/21dee16522bcc9843dd0d18e9bca71bed3081ed4) failed, but the subsequent run based on [89b0e36](https://github.com/lf-lang/reactor-c/commit/89b0e3617db1d2730552a418fd35f1395afe114a) passed, even though that commit only includes comments...

No, this branch points to a branch of limgua-franka, which changed.  So this is not a nondeterministic failure.  Still, I’ve found other problems, so I will convert this PR back to a draft…

File: core/federated/federate.c [hunk 1]
```diff
@@ -623,7 +624,7 @@ static int handle_tagged_message(int* socket, int fed_id) {
     }
 #endif // FEDERATED_DECENTRALIZED
        // The following will update the input_port_action->last_known_status_tag.
-       // For decentralized coordination, this is needed for the thread implementing STAA.
+       // For decentralized coordination, this is needed to unblock the STAA.
     update_last_known_status_on_input_port(env, actual_tag, port_id);
```
[the remark this answers is not in the document]
Comment (edwardalee) [reply core/federated/federate.c:628]: I think this is also used in centralized coordination when there are zero-delay cycles, but I'm not sure. In any case, I think it is harmless to do this even with centralized coordination. The field it updates could be useful for other purposes.

**Review by lhstrh — APPROVED** [review]
LGTM!

**Pull request merged**